### ISSUE TITLE
Output file is produced even if there are errors

### ISSUE THREAD

**Issue by MatrixFrog** [google/closure-compiler#939]
See https://code.google.com/p/chromium/issues/detail?id=486093 -- apparently we are still writing an output file even if there are errors. This might happen only in IDE mode, but we should probably investigate.

**Comment by dimvar** [comment]
IIRC, the compilation in IDE mode continues after errors, so it's not too surprising that it produces an output. We should still fix it though.

**Comment by concavelenz** [comment]
likely we should never produce output for IDE mode.  It might be their runner though.

**Comment by MatrixFrog** [comment]
It looks like we don't produce output in IDE mode so I'm guessing this is actually a bug in Chrome's setup.

**Comment by MatrixFrog** [comment]
^ @danbeam

**Comment by danbeam** [comment]
nah, not [the runner](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/closure_compiler/runner/src/org/chromium/closure/compiler/Runner.java&q=runner.java&sq=package:chromium&type=cs), afaict, as it still happens with just compiler.jar.

I intentionally added an error:

```diff
diff --git a/ui/webui/resources/js/assert.js b/ui/webui/resources/js/assert.js
index 0e20ba5..a6b157a 100644
--- a/ui/webui/resources/js/assert.js
+++ b/ui/webui/resources/js/assert.js
@@ -45,7 +45,7 @@ function assert(condition, opt_message) {
  * This code should only be hit in the case of serious programmer error or
  * unexpected input.
  *
- * @param {string=} opt_message A message to show when this is hit.
+ * @param {string=} message A message to show when this is hit.
  */
 function assertNotReached(opt_message) {
  throw new Error(opt_message || 'Unreachable code hit');
```

removed the old code

    $ rm -rf out/Default/gen/closure/

made the target dir (else I get [a stack](https://gist.githubusercontent.com/danbeam/60f33385b634363d5825/raw/b13d97aecfd699e8c550ac2bfb1b98350005287b/gistfile1.txt) about a missing dir)

    $ mkdir -p out/Default/gen/closure/ui/webui/resources/js/

and ran just compiler.jar

    $ java -jar third_party/closure_compiler/compiler/compiler.jar --accept_const_keyword --language_in=ECMASCRIPT5_STRICT --summary_detail_level=3 --compilation_level=SIMPLE_OPTIMIZATIONS --source_map_format=V3 --polymer_pass --jscomp_error=accessControls --jscomp_error=ambiguousFunctionDecl --jscomp_error=checkStructDictInheritance --jscomp_error=checkTypes --jscomp_error=checkVars --jscomp_error=constantProperty --jscomp_error=deprecated --jscomp_error=externsValidation --jscomp_error=globalThis --jscomp_error=invalidCasts --jscomp_error=missingProperties --jscomp_error=missingReturn --jscomp_error=nonStandardJsDocs --jscomp_error=suspiciousCode --jscomp_error=undefinedNames --jscomp_error=undefinedVars --jscomp_error=unknownDefines --jscomp_error=uselessCode --jscomp_error=visibility --extra_annotation_name=attribute --extra_annotation_name=status --extra_annotation_name=element --extra_annotation_name=homepage --extra_annotation_name=submodule --extra_annotation_name=group --jscomp_off=duplicate --jscomp_off=misplacedTypeAnnotation --js=ui/webui/resources/js/assert.js --js_output_file=out/Default/gen/closure/ui/webui/resources/js/assert.js --create_source_map=out/Default/gen/closure/ui/webui/resources/js/assert.js.map --externs=third_party/polymer/v0_8/components-chromium/polymer-externs/polymer.externs.js
    ui/webui/resources/js/assert.js:50: WARNING - parameter message does not appear in assertNotReached's parameter list
    function assertNotReached(opt_message) {
    ^
    
    0 error(s), 1 warning(s), 87.8% typed

great: I got an error (expected).

not so great: the file was written anyways...

    $ ll out/Default/gen/closure/ui/webui/resources/js/
    total 8
    -rw-r----- 1 user group 294 May 20 15:47 assert.js
    -rw-r----- 1 user group 781 May 20 15:47 assert.js.map

I highly suspect using `--jscomp_error` to upgrade warnings to errors isn't working in all cases.

**Comment by concavelenz** [comment]
@danbeam There is no diagnostic group for the warnings defined in jscomp/FunctionTypeBuilder.java the only way to promote them to errors is with something like the jscomp/StrictWarningsGuard.java which I thought we exposed as something like "--warnings_as_errors" but I don't see it so maybe it never landed.

So there are a couple of options:
1) add the warnings guard with your custom compiler runner 
2) add a diagnostic group for the warnings
3) add a flag
